src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,150 @@
-              <span className="icon" data-cy="iconCompleted">
-                <i className="fas fa-check" />
-              </span>
-            </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-success">
-                illo est ratione doloremque quia maiores aut
-              </p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
+      {loading ? (
+        <Loader />
+      ) : error ? (
+        <p className="notification is-danger">{error}</p>
+      ) : filteredTodos.length === 0 ? (
+        <p className="notification is-warning">
+          There are no todos matching current filter criteria
+        </p>
+      ) : (
+        <table className="table is-narrow is-fullwidth">
+          <thead>
+            <tr>
+              <th>#</th>
+              <th>
                 <span className="icon">
-                  <i className="far fa-eye" />
+                  <i className="fas fa-check" />
                 </span>
-              </button>
-            </td>
-          </tr>
-        </tbody>
-      </table>
+              </th>
+              <th>Title</th>
+              <th> </th>
+            </tr>
+          </thead>
+          <tbody>
+            {filteredTodos.map((todo: Todo) => (
+              <tr key={todo.id} data-cy="todo">
+                <td className="is-vcentered">{todo.id}</td>
+                <td className="is-vcentered">
+                  {todo.completed && (
+                    <span className="icon">
+                      <i className="fas fa-check" />
+                    </span>
+                  )}
+                </td>
+                <td className="is-vcentered is-expanded">
+                  <p
+                    className={
+                      todo.completed ? 'has-text-success' : 'has-text-danger'
+                    }
+                  >
+                    {todo.title}
+                  </p>
+                </td>
+                <td className="has-text-right is-vcentered">
+                  <button
+                    data-cy="selectButton"
+                    className="button"
+                    type="button"
+                    onClick={() => handleSelectTodo(todo)}
+                  >
+                    <span className="icon">
+                      <i
+                        className={
+                          selectedTodoId === todo.id && isModalOpen
+                            ? 'far fa-eye-slash'
+                            : 'far fa-eye'
+                        }
+                      />
+                    </span>
+                  </button>
+                </td>
+              </tr>
+            ))}
+          </tbody>
+        </table>
+      )}
+      {isModalOpen && currentTodo && (
+        <TodoModal onClose={handleCloseModal} user={currentUser} />
+      )}
     </>
   );
```

Comment:
The render function is quite large and complex. Consider breaking it down into smaller, more manageable functions or sub-components. For instance, the table rendering could be moved to a separate component.